tests/model_registry/rbac/conftest.py

```diff
@@ -159,18 +161,116 @@ def model_registry_group_with_user(
         yield group
 
 
+@pytest.fixture(scope="package")
+def user_credentials_rbac() -> dict[str, str]:
+    random_str = generate_random_name()
+    return {
+        "username": f"test-user-{random_str}",
+        "password": f"test-password-{random_str}",
+        "idp_name": f"test-htpasswd-idp-{random_str}",
+        "secret_name": f"test-htpasswd-secret-{random_str}",
+    }
+
+
 @pytest.fixture(scope="session")
-def test_idp_user_session() -> Generator[UserTestSession, None, None]:
+def original_user() -> str:
+    current_user = run_command(command=["oc", "whoami"])[1].strip()
+    LOGGER.info(f"Original user: {current_user}")
+    return current_user
+
+
+@pytest.fixture(scope="module")
+def created_htpasswd_secret(
+    original_user: str, user_credentials_rbac: dict[str, str]
+) -> Generator[UserTestSession, None, None]:
+    """
+    Session-scoped fixture that creates a test IDP user and cleans it up after all tests.
+    Returns a UserTestSession object that contains all necessary credentials and contexts.
+    """
+
+    temp_path, htpasswd_b64 = create_htpasswd_file(
+        username=user_credentials_rbac["username"], password=user_credentials_rbac["password"]
+    )
+    try:
+        LOGGER.info(f"Creating secret {user_credentials_rbac['secret_name']} in openshift-config namespace")
+        with Secret(
+            name=user_credentials_rbac["secret_name"],
+            namespace="openshift-config",
+            htpasswd=htpasswd_b64,
+            type="Opaque",
+            wait_for_resource=True,
+        ) as secret:
+            yield secret
+    finally:
+        # Clean up the temporary file
+        temp_path.unlink(missing_ok=True)
+
+
+@pytest.fixture(scope="module")
+def updated_oauth_config(
```

Comment:
same question as above

Reply:
Same response. This needs to be updated or it would introduce another bug that would be uncovered if these needs to be parameterized.